index.js

```diff
@@ -5,18 +5,28 @@ const fs = require('fs');
 
 app.use(express.static("./public"));
 
+let errorsWhileGettingData;
+
 const fetchData = async () => {
-  const response = await axios.get("https://diablo2.io/dclone_api.php?", {
-    params: {
-      sk: "r",
-    },
-  });
-  allRegions = response.data;
-  // Faking response for testing purposes:
-  // allRegions[0].reporter_id = 333
-  // allRegions[0].progress = 5
-  // allRegions[10].progress = 9
-  return allRegions;
+
+  errorsWhileGettingData ?? null;
+
+  try {
```

Comment:
It seems that the main purpose of this function is to make an API call and return some data. So I'd expect it to throw an error if `axios.get` throws an error. I don't think that we need `try/catch` block here, I'd leave it as is:
```js
const fetchData = async () => {
  const response = await axios.get("https://diablo2.io/dclone_api.php?", {
    params: {
      sk: "r",
    },
  });
  const allRegions = response.data;
  return allRegions;
}
```

On the other end, code that **uses this function** should know what to do in case of an error coming from `fetchData()` call. So we can use `try/catch` block in `collectDataOnceInAMinute`, something like this:
```js
async function collectDataOnceInAMinute() {
  let dataForLogging = null;
  try {
    dataForLogging = await fetchData();
    currentData = dataForLogging;
  } catch (error) {
    dataForLogging = error;
  }
  createLogs(dataForLogging);
  setTimeout(() => {
    collectDataOnceInAMinute();
  }, 30000);
}
```